### PR TITLE
Report finished

### PR DESCRIPTION
**Pull request by Vedo-Kodecta**

### CODE REVIEW

File: app/Http/Controllers/BaseController.php [hunk 1]
```diff
@@ -69,4 +70,20 @@ public function destroy(int $id)
...
+    {
+        $filePath = $request->query('filePath');
+        $fileName = basename($filePath);
+
+        if (!$filePath) {
+            abort(404);
+        }
+
+        if (!file_exists($filePath)) {
+            abort(404);
```
Comment (ErminMaksumic) [inline app/Http/Controllers/BaseController.php]: This can be one if statement.

File: react/src/app/context/Product/api.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { Product } from "@/lib/product";
 import { Variant } from "@/lib/variant";
+import { daDK } from "@mui/x-data-grid";
```
Comment (ErminMaksumic) [inline react/src/app/context/Product/api.ts]: Remove this import.

**Review by ErminMaksumic — CHANGES_REQUESTED** [review]

File: routes/api.php [hunk 3]
```diff
@@ -25,6 +26,8 @@
 Route::post('/product/{id}/generateReport', [ProductController::class, 'generateReportForOneProduct']);
 Route::post('/product/generateReport', [ProductController::class, 'generateReportForExpensiveProducts']);
 Route::post('/product/generateReportChart', [ProductController::class, 'generateReportForProductStatesGraph']);
+Route::get('/download', [ProductController::class, 'download'])->name('download');
+Route::get('/deleteFile', [ProductController::class, 'deleteFile'])->name('deleteFile');
```
Comment (ErminMaksumic) [inline routes/api.php]: Do we need this route?

**Review by ErminMaksumic — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/BaseController.php (hunk 1 above)
Comment (Vedo-Kodecta) [reply app/Http/Controllers/BaseController.php]: fixed please can u check again

File: react/src/app/context/Product/api.ts (hunk 2 above)
Comment (Vedo-Kodecta) [reply react/src/app/context/Product/api.ts]: removed

File: routes/api.php (hunk 3 above)
Comment (Vedo-Kodecta) [reply routes/api.php]: This was old code, removed it. Can you check again

**Review by ErminMaksumic — APPROVED** [review]
lgtm now

**Pull request merged**